cmd/helm/upgrade.go

```diff
@@ -74,27 +74,14 @@ func newUpgradeCmd(cfg *action.Configuration, out io.Writer) *cobra.Command {
 		RunE: func(cmd *cobra.Command, args []string) error {
 			client.Namespace = settings.Namespace()
 
-			if client.Version == "" && client.Devel {
-				debug("setting version to >0.0.0-0")
-				client.Version = ">0.0.0-0"
-			}
-
-			vals, err := valueOpts.MergeValues(getter.All(settings))
-			if err != nil {
-				return err
-			}
-
-			chartPath, err := client.ChartPathOptions.LocateChart(args[1], settings)
-			if err != nil {
-				return err
-			}
-
+			// Fixes #7002 - Support reading values from STDIN for `upgrade` command
+			// Must load values AFTER determining if we have to call install so that values loaded from stdin are are not read twice
 			if client.Install {
 				// If a release does not exist, install it. If another error occurs during
 				// the check, ignore the error and continue with the upgrade.
 				histClient := action.NewHistory(cfg)
 				histClient.Max = 1
-				if _, err := histClient.Run(args[0]); err == driver.ErrReleaseNotFound {
+				if release, err := histClient.Run(args[0]); err == driver.ErrReleaseNotFound || release == nil {
```

Comment:
why is the check for `release == nil` necessary?

Reply:
The "memory" driver used in test cases seems not not behave identically to the regular driver.  It's returning nil when there is no history instead of throwing the driver.ErrReleaseNotFound error.  I thought it was lower risk to just check for nil here then change the behavior of it.  Thoughts?

Reply:
I actually decided to go and fix the memory driver so it behaved like the other drivers and ended up discovering a defect in the rollback functionality.  When a rollback occurred and it rolled back to a release in which there were no prior deployed releases, it would cause a "... has no deployed releases" error.  I fixed that defect with my latest commit.